keps/sig-scheduling/2249-pod-affinity-namespace-selector/README.md

```diff
@@ -0,0 +1,663 @@
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+By default, pod affinity/anti-affinity constraints are calculated against 
+the pods in the same namespace. The Spec allows users to exapnd that via
+the `PodAffinityTerm.Namespaces` list. 
+
+This KEP proposes adding NamespaceSelector to `PodAffinityTerm` so that users 
+can specify the set of namespace that should considered using common labels applied
+to relevant namespaces.
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+Pod affinity/anti-affinity API allows using this feature across namespaces using 
+a static list, which works if the user knowns the namespaces names in advance.
+
+However, there are cases where the namespaces are not known beforehand, for which there
+is no way to use pod affinity/anti-affinity. Allowing users to specify the set of namespaces
+using a namespace selector addresses this problem.
```

Comment:
affinity/anti-affinity already allows enumerating namespaces by name... I don't see how #2162 helps here...

Reply:
Is it important to select a subset of namespaces, or would it be sufficient to allow affinity/anti-affinity to apply across all namespaces (effectively making the pod labels globally meaningful)? That would be a simpler API surface and would remove the need to run label queries on namespaces

Reply:
links to previous discussions of the implications of this would be helpful (https://github.com/kubernetes/kubernetes/issues/68827#issuecomment-468470598, https://github.com/kubernetes/kubernetes/issues/43203#issuecomment-287237992)

The two main categories are performance (does this let a single pod make the scheduler do lots of expensive work) and security (does this let a badly behaved namespace disrupt other namespaces)